### PR TITLE
Update spec to reflect recent changes

### PR DESCRIPTION
**Pull request by dorranh**
Updates the documentation and functional spec to reflect the recent generalization work which has resulted in some changes to the descriptions in Checker's functional spec and the addition of two new contracts to the project.

### CODE REVIEW

**Comment by dorranh** [comment]
Marking as ready for review. There are still probably some areas where we can elaborate, but this at least updates the functional spec and intro page.

File: docs/spec/introduction.rst [hunk 1]
```diff
@@ -47,34 +47,69 @@ System overview of a Checker deployment
...
+Checker currently supports the following variations:
+
+1. Collateral type
+
+   a. ``collateral_type=fa2`` - In this case, the collateral type is an existing
+      FA2 token and the CFMM is configured to use the same FA2 token.
+
+   b. ``collateral_type=tez`` - A special case of (1). In this case Checker's CFMM is
+      required to use `ctez` and the FA2 collateral contract must use the `wtez` FA2
+      wrapper for `tez` which is included in the Checker repository.
```
Comment (gkaracha) [inline docs/spec/introduction.rst:106]:
```suggestion
      required to use the `wctez` FA2 wrapper for `ctez` and the FA2 collateral
      contract must use the `wtez` FA2 wrapper for `tez`, both of which are included
      in the Checker repository.
```

File: docs/spec/functional_spec.rst [hunk 2]
```diff
@@ -608,3 +640,292 @@ Seal the contract and make it ready for use
...
++---------------+-----------------------+-------------------------------------------------------------------------+
+| amount        | mutez                 | The amount of mutez to send from the main contract to the vault         |
++---------------+-----------------------+-------------------------------------------------------------------------+
+
+
+Send tez from a vault to a contract
+-----------------------------------
+
+Internal. Calls the specified vault's ``%vault_send_tez_to_contract``
+entrypoint, sending the specified amount of ``mutez``. from the vault to the
```
Comment (gkaracha) [inline docs/spec/functional_spec.rst:780]:
```suggestion
entrypoint, sending the specified amount of ``mutez`` from the vault to the
```

File: docs/spec/functional_spec.rst [hunk 3]
```diff
@@ -608,3 +640,292 @@ Seal the contract and make it ready for use
...
++===============+=======================+=========================================================================+
+| vault_address | address               | The address of the vault                                                |
++---------------+-----------------------+-------------------------------------------------------------------------+
+| amount        | tez                   | The amount of mutez to send from the main contract to the vault         |
++---------------+-----------------------+-------------------------------------------------------------------------+
+| recipient     | address               | The address of the vault which will receive the mutez                   |
++---------------+-----------------------+-------------------------------------------------------------------------+
+
+Set a vault's optional delegate
+-------------------------------
```
Comment (gkaracha) [inline docs/spec/functional_spec.rst:816]:
```suggestion
Set a vault's delegate
----------------------
```
it's pretty standard, the fact that this is an optional so it should be good like this

**Review by gkaracha — APPROVED** [review]
Nice! :ok_hand: This is certainly an improvement over what we had before, I only left a couple small suggestions. We can merge this and iterate and fill-in the `TODO`s later (not all parts missing have been implemented yet anyway).

**Pull request merged**